src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,255 @@
+
+type Props = {
+  budget: Budget
+  setNotification: (notification: string) => void
+}
+
+const MatchRuleList = ({ budget, setNotification }: Props) => {
+  const { t }: Translation = useTranslation()
+
+  const [error, setError] = useState('')
+  const [isLoading, setIsLoading] = useState(true)
+  const [hideHelp, setHideHelp] = useState(true)
+
+  const [matchRules, setMatchRules] = useState<MatchRule[]>([])
+  const [accounts, setAccounts] = useState<Account[]>([])
+  const accountGroups = [
+    {
+      title: t('accounts.internalAccounts'),
+      items: accounts.filter(account => !account.external),
+    },
+    {
+      title: t('accounts.externalAccounts'),
+      items: accounts.filter(account => account.external),
+    },
+  ]
+
+  const addMatchRule = () => {
+    // The ID we set here is needed for uniqueness in the sortable list, but ignored later on when we actually send the data to the backend.
+    // The priority is in order from top to bottom and can be 0, so we can just use the length of the matchRules.
+    //
+    // We also set the links to a dummy value since we only need them for match rules loaded from the backend
+    setMatchRules([
+      { id: matchRules.length.toString(), links: {} },
+      ...matchRules,
+    ])
+  }
+
+  const deleteMatchRule = (id: string) => {
+    console.log(id)
+    let newMatchRules = [...matchRules]
+    const ruleIndex = matchRules.findIndex(rule => rule.id === id)
+
+    newMatchRules[ruleIndex].toDelete = true
+    console.log(newMatchRules)
+    setMatchRules(newMatchRules)
+  }
+
+  const saveMatchRules = async () => {
+    let updatedMatchRules: MatchRule[] = []
+
+    const requests = matchRules.map((rule, index) => {
+      // If the rule should be deleted, we delete it. This is done before checking validity, because
+      // a rule might have been edited to be invalid and then deleted
+      //
+      // Rules that have not been created yet are also skipped, since they will be gone with the re-render
+      if (rule.toDelete && rule.createdAt) {
+        return matchRuleApi.delete(rule).catch(err => setError(err.message))
+      }
+
+      // If the rule is missing an account ID or a match, it is invalid, so we discard it
+      if (!rule.accountId || !rule.match) {
+        return true
+      }
+
+      if (rule.createdAt) {
+        // createdAt is set by the backend, so this is an existing rule that needs to be updated
+        rule.priority = index
+
+        return matchRuleApi
+          .update(rule)
+          .then(updatedRule => {
+            updatedMatchRules.push(updatedRule)
+          })
+          .catch(err => setError(err.message))
+      } else {
+        // This is a new rule, it needs to be created
+        rule.priority = index
+        return matchRuleApi
+          .create(rule, budget)
+          .then(newRule => {
+            updatedMatchRules.push(newRule)
+          })
+          .catch(err => setError(err.message))
```

Comment:
This way, errors overwrites any previous errors. Maybe we should have an `errors` array that we push the message into and then, if it is not empty in the end, display them all.

Reply:
Yup! See 72738d7.